rclpy/src/rclpy/_rclpy.c

```diff
@@ -636,15 +638,17 @@ rclpy_create_publisher(PyObject * Py_UNUSED(self), PyObject * args)
   *publisher = rcl_get_zero_initialized_publisher();
   rcl_publisher_options_t publisher_ops = rcl_publisher_get_default_options();
 
+  if (!PyCapsule_CheckExact(pyqos_profile)) {
+    PyErr_Format(PyExc_RuntimeError, "Invalid pyqos_profile capsule in rclpy_create_publisher");
+    return NULL;
+  }
```

Comment:
Doesn't this change semantic? Before it was possible to not pass a profile.

Same below.

Reply:
Oops I thought it was a bug not a feature XD. I'll restore the previous behavior then. Note that we do also ensure a qos_profile is passed to these functions at the rclpy Python level, but I agree it's better to keep it flexible at the C-level as well